example-library/src/main/java/uk/gov/gchq/palisade/example/config/ExamplePolicyCacheWarmerFactory.java

```diff
@@ -0,0 +1,194 @@
+import uk.gov.gchq.palisade.example.rule.NationalityRule;
+import uk.gov.gchq.palisade.example.rule.RecordMaskingRule;
+import uk.gov.gchq.palisade.example.rule.ZipCodeMaskingRule;
+import uk.gov.gchq.palisade.service.PolicyCacheWarmerFactory;
+import uk.gov.gchq.palisade.service.request.Policy;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import static java.util.Objects.requireNonNull;
+
+@ConfigurationProperties
+public class ExamplePolicyCacheWarmerFactory implements PolicyCacheWarmerFactory {
+
+    private String type;
+    private String owner;
+    private Map<String, String> resourceRules;
+    private Map<String, String> recordRules;
+
+    /**
+     * Constructor with 0 arguments for an example implementation
+     * of the {@link PolicyCacheWarmerFactory} interface
+     */
+    public ExamplePolicyCacheWarmerFactory() {
+    }
+
+    /**
+     * Constructor with 4 arguments for an example implementation
+     * of the {@link PolicyCacheWarmerFactory} interface
+     *
+     * @param type          a {@link String} of the {@link Policy} type.
+     * @param owner         a {@link String} value of the owner of the policy
+     * @param resourceRules a {@link Map} containing the ({@link String}) message and the ({@link String}) rule name.
+     * @param recordRules   a {@link Map} containing the ({@link String}) message and the ({@link String}) rule name.
+     */
+    public ExamplePolicyCacheWarmerFactory(final String type, final String owner, final Map<String, String> resourceRules, final Map<String, String> recordRules) {
+        this.type = type;
+        this.owner = owner;
+        this.resourceRules = resourceRules;
+        this.recordRules = recordRules;
+    }
+
+    @Generated
+    public String getType() {
+        return type;
+    }
+
+    @Generated
+    public void setType(final String type) {
+        requireNonNull(type);
+        this.type = type;
+    }
+
+    @Generated
+    public String getOwner() {
+        return owner;
+    }
+
+    @Generated
+    public void setOwner(final String owner) {
+        requireNonNull(owner);
+        this.owner = owner;
+    }
+
+    @Generated
+    public Map<String, String> getResourceRules() {
+        return resourceRules;
+    }
+
+    @Generated
+    public void setResourceRules(final Map<String, String> resourceRules) {
+        requireNonNull(resourceRules);
+        this.resourceRules = resourceRules;
+    }
+
+    @Generated
+    public Map<String, String> getRecordRules() {
+        return recordRules;
+    }
+
+    @Generated
+    public void setRecordRules(final Map<String, String> recordRules) {
+        requireNonNull(recordRules);
+        this.recordRules = recordRules;
+    }
+
+    @Override
+    public Policy<Employee> policyWarm() {
+        Policy<Employee> policy = new Policy<>();
+        if ("Alice".equals(getOwner())) {
+            policy.owner(ExampleUsers.getAlice());
```

Comment:
This has the risk of the hardcoded user details being out of sync with the config user details loaded by the user cache warming. This would be much better as a call off to the user service to get the user with the given uid. Given that would require a new API to be created for the user service I think it would be best to have this as a full user info pulled from the policy service caching yaml. There will need to be a note in both config yaml files to update the other of changes

Reply:
Users have been added to the PolicyConfiguration, created from the yaml, and are passed to the policyWarm method so there is no need to use the ExampleUsers class.